src/Sulu/Bundle/MediaBundle/Resources/js/stores/MediaUploadStore/MediaUploadStore.js

```diff
@@ -41,26 +23,46 @@ export default class MediaUploadStore {
         this.progress = Math.ceil(progress);
     }
 
-    update(mediaId: string | number, file: File) {
+    @action setData(data: Object) {
+        this.data = data;
+    }
+
+    update(mediaId: string | number, file: File): Promise<*> {
         const baseUrl = ResourceMetadataStore.getBaseUrl(RESOURCE_KEY);
         const queryString = ResourceRequester.buildQueryString({
             action: 'new-version',
-            locale: this.resourceStore.locale.get(),
+            locale: this.locale.get(),
         });
         const url = baseUrl + '/' + mediaId + queryString;
 
         this.setUploading(true);
 
-        this.upload(file, url)
-            .then((data: Object) => {
-                for (const key of Object.keys(data)) {
-                    this.resourceStore.set(key, data[key]);
```

Comment:
How is the `resourceStore` updated now? Or is that not necessary anymore for some reason?